File: stage1/s1_if_03.js
// S1_IF_03 — Truthy / falsy guard
function normalizeName(input) {
  if (!input) {
    return "Anonymous";
  }
  return String(input).trim();
}

// Tests
console.log(normalizeName("")); // "Anonymous"
console.log(normalizeName(" ")); // ""  (truthy string, then trimmed)
console.log(normalizeName(null)); // "Anonymous"
console.log(normalizeName(" Ola ")); // "Ola"

File: stage1/B/S1_IF_03.py
# S1_IF_03.py

def normalize_name(x):
    if not x:
        return "Anonymous"
    s = str(x).strip()
    return s if s else "Anonymous"


def main():
    tests = ["", " ", None, " Ola "]
    for t in tests:
        print(f"input={t!r:<7} -> {normalize_name(t)!r}")


if __name__ == "__main__":
    main()
